program/src/matching.rs

```diff
@@ -1610,3 +1782,46 @@ impl<'a> Book<'a> {
         Ok(())
     }
 }
+
+fn determine_ref_vars<'a>(
+    program_id: &Pubkey,
+    mango_group: &MangoGroup,
+    mango_group_pk: &Pubkey,
+    mango_cache: &MangoCache,
+    mango_account: &MangoAccount,
+    referrer_mango_account_ai: Option<&'a AccountInfo>,
+    now_ts: u64,
+) -> MangoResult<(I80F48, Option<RefMut<'a, MangoAccount>>)> {
+    let mngo_index = match mango_group.find_token_index(&mngo_token::id()) {
+        None => return Ok((ZERO_I80F48, None)),
+        Some(i) => i,
+    };
+
+    let mngo_cache = &mango_cache.root_bank_cache[mngo_index];
+    let mngo_deposits = mango_account.get_native_deposit(mngo_cache, mngo_index)?;
+    let ref_mngo_req = I80F48::from_num(mango_group.ref_mngo_required);
+    if mngo_deposits >= ref_mngo_req {
+        return Ok((ZERO_I80F48, None));
+    } else if let Some(referrer_mango_account_ai) = referrer_mango_account_ai {
+        if let Ok(referrer_mango_account) =
+            MangoAccount::load_mut_checked(referrer_mango_account_ai, program_id, mango_group_pk)
+        {
+            // Need to check if it's valid because user may not have mngo in active assets
+            mngo_cache.check_valid(mango_group, now_ts)?;
```

Comment:
I was wondering if it's intentional that an error from loading the referrer's MangoAccount is ignored above, but an error in this line or the get_native_deposit() below would be critical. It makes sense though: we don't want users to be affected when referrer closes their account.

Maybe add comments about this?

Reply:
error in get_native_deposit would just be a checked mul error which should always be critical. Error in cache I suppose shouldn't be critical if referrer doesn't have any MNGO. But cache error shouldn't be happening frequently anyway

Reply:
I could move the cache check into the final if block... small potatoes either way